functions/go/set-image/README.md

```diff
@@ -0,0 +1,118 @@
+# set-image
+
+## Overview
+
+<!--mdtogo:Short-->
+
+The `set-image` function sets the image for all instances of a provided image
+`name` with the provided image `newName:newTag` or `newName@digest`. A common
+deployment pattern is to update images to the latest tags or, ideally, digests.
+Sometimes people want to change the image repo, also, such as for dev builds.
+
+<!--mdtogo-->
+
+You can learn more about images [here][image].
+
+<!--mdtogo:Long-->
+
+## Usage
+
+This function can be used with any KRM function orchestrators (e.g. kpt).
+
+For all resources which specify an image of a given `name`, the `set-image`
+function sets that image field to the specified `newName:newTag` or
+`newName@digest`.
+
+By default the function updates the standard [fields][commonimage] used to
```

Comment:
Should we update the images in Kptfile?

Reply:
I am not sure whether or not that would be a desirable default behavior

The user could of course always supply this as an option in `additionalImageFields`, although that would be less intuitive.

I could also see an argument for a separate function specifically for the use case of updating Kptfiles. I could see this as a function which automatically discovers the latest patch version for images specified in Kptfiles and update them dynamically based on what was fetched

Reply:
> I could also see an argument for a separate function specifically for the use case of updating Kptfiles. I could see this as an imperative function which automatically discovers the latest patch version for images specified in Kptfiles and update them dynamically based on what was fetched

No doubt of that. But IMO they are not mutually exclusive. The set-image function can be a supplement to help users to achieve some one-off operations that can't be done by the update-to-latest-image-in-kptfile function. People may argue that one-off operations can be done by manual in-place edit.

If we support updating images in Kptfile in this PR, user will need to specify `eval --include-meta-resources` to make it work, since eval doesn't include Kptfile by default and render doesn't support it.

I don't feel very strong about it and I'd like to hear others' opinions. @droot @phanimarupaka Thoughts?

Reply:
I think extending the set-image logic to update images in `Kptfile` sounds like a good enhancement. But we can address this as a follow-up.